### PR TITLE
Azure SQL Server auto-discovery documentation

### PR DESCRIPTION
**Pull request by gabrielcorado**
Adds the documentation for Azure SQL Server auto-discovery implemented at #17745.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/database-access/guides/azure-sql-server-ad.mdx [hunk 1]
```diff
@@ -125,30 +126,106 @@ $ teleport db configure create \
...
+  group `/providers/Microsoft.Management/managementGroups/<group>` instead.
 </Admonition>
 
+Now go to the [Subscriptions](https://portal.azure.com/#view/Microsoft_Azure_Billing/SubscriptionsBlade)
+page and select a subscription.
+
+Click on **Access control (IAM)** in the subscription and select **Add** >
+**Add custom role**:
+
+<Figure align="left">
```
Comment (alexfornuto) [inline docs/pages/database-access/guides/azure-sql-server-ad.mdx:199]: These images are large enough that they fill the entire space of the content pane in our docs, so the use of the Figure component to align them is superfluous. It's not doing any harm, but you could remove them if you want.

File: docs/pages/database-access/guides/azure-sql-server-ad.mdx [hunk 2]
```diff
@@ -113,10 +117,7 @@ $ teleport db configure create \
    -o file \
    --token=/tmp/token \
    --proxy=teleport.example.com:3080 \
```
[the remark this answers is not in the document]
Comment (gabrielcorado) [reply docs/pages/database-access/guides/azure-sql-server-ad.mdx:122]: I see that in some [examples](https://goteleport.com/docs/connect-your-client/introduction/?scope=cloud#client-tools), the command example only points to the domain using the default HTTPS port. Is this what you're suggesting? Changing it to `--proxy=telepot.example.com`.

[the remark this answers is not in the document]
Comment (alexfornuto) [reply docs/pages/database-access/guides/azure-sql-server-ad.mdx:122]: Yea I guess so. We don't know if the reader already has a convention set to connect to auth or proxy service, but using the proxy service is the default example we want to set moving forward.

File: docs/pages/database-access/guides/azure-sql-server-ad.mdx [hunk 3]
```diff
@@ -99,77 +100,152 @@ The newly created user will be attached to the public role, which might not
...
 
 <ScopedBlock scope={["oss", "enterprise"]}>
+
 ```code
 $ tsh login --proxy=teleport.example.com --user=alice
 $ tsh db ls
 # Name      Description         Labels
 # --------- ------------------- -------
 # sqlserver                     env=dev
 ```
```
Comment (greedy52) [inline docs/pages/database-access/guides/azure-sql-server-ad.mdx:236]: Could we give an example (or examples) of the discovered SQL server with populated descriptions and labels?

**Review by greedy52 — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]
👍 , but the outstanding comment in the previous review might still be worth adjusting for.

**Review by smallinsky — APPROVED** [review]

**Pull request merged**